src/components/cylc/tree/deltas.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  createCyclePointNode,
+  createFamilyProxyNode,
+  createJobNode,
+  createTaskProxyNode, populateTreeFromGraphQLData
+} from '@/components/cylc/tree/index'
+// eslint-disable-next-line no-unused-vars
+import CylcTree from './tree'
+
+/**
+ * Deltas pruned.
+ *
+ * @param {{
+ *   taskProxies: [string],
+ *   familyProxies: [string],
+ *   jobs: [string]
+ * }} pruned
+ * @param {CylcTree} tree
+ */
+function applyDeltasPruned (pruned, tree) {
+  // jobs
+  if (pruned.jobs) {
```

Comment:
It might be a nice to reduce these branches to a mapping e.g:

```js
PRUNED = {
    'jobs': 'removeJob',
    'taskProxies': 'removeTaskProxy',
   // ...
}
```

Just to simplify and formalise the solution and prevent surplus logic getting added to these functions.

Reply:
@oliver-sanders updated! Only downside of the change is that I lost ability to trace/refactor the functions `removeJob`, `createJobNode`, etc, from within the IDE. As we are using strings, and the IDE is not clever enough to identify the function usage.

I can still search-by-string, and with enough tests it should be OK I think?